service/src/main/java/bio/terra/pipelines/service/PipelineRunsService.java

```diff
@@ -0,0 +1,187 @@
+
+    if (pipeline.getWorkspaceId() == null) {
+      throw new InternalServerErrorException("%s workspaceId not defined".formatted(pipelineName));
+    }
+
+    PipelineRun pipelineRun =
+        writePipelineRunToDb(
+            jobId,
+            userId,
+            pipeline.getId(),
+            CommonPipelineRunStatusEnum.SUBMITTED,
+            description,
+            resultPath,
+            userProvidedInputs);
+
+    logger.info("Creating new {} job for user {}", pipelineName, userId);
+
+    Class<? extends Flight> flightClass;
+    switch (pipelineName) {
+      case IMPUTATION_BEAGLE:
+        flightClass = RunImputationJobFlight.class;
+        break;
+      default:
+        throw new InternalServerErrorException(
+            "Pipeline %s not supported by PipelineRunsService".formatted(pipelineName));
+    }
+
+    JobBuilder jobBuilder =
+        jobService
+            .newJob()
+            .jobId(jobId)
+            .flightClass(flightClass)
+            .addParameter(JobMapKeys.PIPELINE_NAME.getKeyName(), pipelineName)
+            .addParameter(JobMapKeys.USER_ID.getKeyName(), userId)
+            .addParameter(JobMapKeys.DESCRIPTION.getKeyName(), description)
+            .addParameter(RunImputationJobFlightMapKeys.PIPELINE_ID, pipeline.getId())
+            .addParameter(
+                RunImputationJobFlightMapKeys.PIPELINE_INPUT_DEFINITIONS,
+                pipeline.getPipelineInputDefinitions())
+            .addParameter(
+                RunImputationJobFlightMapKeys.USER_PROVIDED_PIPELINE_INPUTS, userProvidedInputs)
+            .addParameter(
+                RunImputationJobFlightMapKeys.CONTROL_WORKSPACE_ID,
+                pipeline.getWorkspaceId().toString())
+            .addParameter(
+                RunImputationJobFlightMapKeys.WDL_METHOD_NAME, pipeline.getWdlMethodName())
+            .addParameter(JobMapKeys.RESULT_PATH.getKeyName(), resultPath);
+
+    jobBuilder.submit();
+
+    logger.info("Created {} pipelineRun with jobId {}", pipelineName, jobId);
+
+    return pipelineRun;
+  }
+
+  public PipelineRun writePipelineRunToDb(
+      UUID jobUuid,
+      String userId,
+      Long pipelineId,
+      CommonPipelineRunStatusEnum status,
+      String description,
+      String resultUrl,
+      Map<String, Object> pipelineInputs) {
+
+    // write pipelineRun to database
+    PipelineRun pipelineRun =
+        new PipelineRun(jobUuid, userId, pipelineId, status.toString(), description, resultUrl);
+    PipelineRun createdPipelineRun = writePipelineRunToDbThrowsDuplicateException(pipelineRun);
+
+    String pipelineInputsAsString;
+    try {
+      // do this to write the pipeline inputs without writing the class name
+      pipelineInputsAsString = objectMapper.writeValueAsString(pipelineInputs);
+    } catch (JsonProcessingException e) {
+      throw new InternalServerErrorException(
+          "THIS SHOULD NEVER HAPPEN! Error converting pipeline inputs to string", e);
```

Comment:
i know this was in there before the chagne but this is probably better left as a log message and not soemthing we might return to the user 😂 .  Probably better to have a differ exception message